warehouse/core/src/main/java/datawave/iterators/filter/ageoff/LatestVersionFilter.java

```diff
@@ -0,0 +1,396 @@
+ *      If a dataType is configured to be filtered but has no configured mode, it will default to reject all data for Scan and accept all data for Compaction
+ *      scopes.
+ *
+ */
+public class LatestVersionFilter extends Filter {
+    private static final Logger log = Logger.getLogger(LatestVersionFilter.class);
+    
+    private static final String IS_INDEX_TABLE_PROPERTY_NAME = "table.custom." + AgeOffConfigParams.IS_INDEX_TABLE;
+    static final String DATATYPE_LIST_OPTION_NAME = "dataTypes";
+    static final String DATATYPE_MODE_OPTION_NAME = "mode";
+    static final String DATATYPE_TIMESTAMP_PROPERTY_NAME_PREFIX = "table.custom.timestamp.current.";
+    static final String IS_INDEX_TABLE_OPTION_NAME = AgeOffConfigParams.IS_INDEX_TABLE;
+    
+    // Index table keys are parsed differently to extract the DataType
+    @VisibleForTesting
+    boolean isIndexTable;
+    
+    // Maps: DataType -> Configuration for filtering logic
+    @VisibleForTesting
+    Map<ByteSequence,VersionFilterConfiguration> dataTypeConfigurations = null;
+    
+    // Iterator Scope used only in failure case to protect data
+    @VisibleForTesting
+    boolean isScanScope;
+    
+    public LatestVersionFilter() {
+        super();
+    }
+    
+    /**
+     * @param key
+     *            parsed for dataType, timestamp examined
+     * @param value
+     *            ignored
+     * @return true if the key's timestamp conforms to the filterConfiguration for its datatype
+     */
+    @Override
+    public boolean accept(Key key, Value value) {
+        // parse dataType from key
+        ByteSequence dataType = DataTypeParser.parseKey(key, this.isIndexTable);
+        
+        // determine if dataType is configured for filtering
+        VersionFilterConfiguration filterConfiguration = dataTypeConfigurations.get(dataType);
+        
+        // if dataType is not configured for filtering, just accept the key / value
+        if (null == filterConfiguration) {
+            return true;
+        }
+        
+        // evaluate key's timestamp version against configuration
+        return accept(key.getTimestamp(), filterConfiguration);
+    }
+    
+    /**
+     * @param dataTimestamp
+     *            the timestamp version for the key under evaluation
+     * @param filterConfiguration
+     *            a non-null filterConfiguration
+     * @return whether to accept or reject this timestamp
+     */
+    @VisibleForTesting
+    boolean accept(long dataTimestamp, VersionFilterConfiguration filterConfiguration) {
+        // determine dataType's configured timestamp
+        long configuredTimestamp = filterConfiguration.timestampVersion;
+        
+        // apply logic for configured mode
+        switch (filterConfiguration.mode) {
+            case EQUAL:
+                return dataTimestamp == configuredTimestamp;
+            case GREATER_THAN_OR_EQUAL:
+                return dataTimestamp >= configuredTimestamp;
+            case UNDEFINED:
```

Comment:
What is the purpose of the UNDEFINED switch case?  Should an exception be raised, or maybe a warning logged?

Reply:
Sometimes table iterator options are modified and I'd rather not throw exceptions.  It's outputting a log.warn.  The code handles the situation where it's undefined and have sensible behavior: accept all data for compaction scopes (to avoid data loss) and hide all data for scan (to avoid accidentally accepting data intended to be rejected).

Reply:
I think this will flood the logs though so I'm thinking it makes more sense to only log when the datatype's configuration is read.  I'll make that change.